contracts/escrows/CountdownGriefingEscrow.sol

```diff
@@ -0,0 +1,357 @@
+
+        // create the agreement
+
+        address agreement;
+        {
+            // get the agreement factory address
+
+            address escrowFactory = Template.getFactory();
+            address escrowRegistry = iFactory(escrowFactory).getInstanceRegistry();
+            address agreementFactory = abi.decode(iRegistry(escrowRegistry).getFactoryData(escrowFactory), (address));
+
+            // encode initialization function
+            bytes memory initCalldata = abi.encodeWithSignature(
+                'initialize',
+                address(this),
+                _data.seller,
+                _data.buyer,
+                uint256(_data.agreementParams.ratio),
+                _data.agreementParams.ratioType,
+                uint256(_data.agreementParams.countdownLength),
+                bytes("0x")
+            );
+
+            // deploy and initialize agreement contract
+            agreement = iFactory(agreementFactory).create(initCalldata);
+        }
+
+        // transfer stake and payment to the agreement
+
+        uint256 totalStake;
+        {
+            if (uint256(_data.paymentAmount) != uint256(0))
+                Staking._removeFullStake(_data.buyer);
+            if (uint256(_data.stakeAmount) != uint256(0))
+                Staking._removeFullStake(_data.seller);
+            totalStake = uint256(_data.paymentAmount).add(uint256(_data.stakeAmount));
+        }
+
+        CountdownGriefing(agreement).increaseStake(0, totalStake);
+
+        // start agreement countdown
+
+        CountdownGriefing(agreement).startCountdown();
+
+        // disable operator
+
+        CountdownGriefing(agreement).renounceOperator();
+
+        // update status
+        _data.status = EscrowStatus.isFinalized;
+
+        // emit event
+        emit Finalized();
+    }
+
+    /// @notice Submit data to the buyer
+    /// @dev Access Control: seller OR operator
+    ///      State Machine: after finalize()
+    function submitData(bytes memory data) public {
+        // restrict access control
+        require(isSeller(msg.sender) || Operated.isActiveOperator(msg.sender), "only seller or active operator");
+        // restrict state machine
+        require(isDeposited(), "only after deposit");
+
+        // emit event
+        emit DataSubmitted(data);
+    }
+
+    /// @notice Finalize escrow and execute completion script
+    ///          - return stake to seller
+    ///          - return payment to buyer
+    ///          - close escrow
+    /// @dev Access Control: seller OR buyer OR operator
+    ///      State Machine: before depositStake() OR before depositPayment() OR after Countdown.isOver()
+    function cancel() public {
+        // restrict access control
+        require(isSeller(msg.sender) || Operated.isActiveOperator(msg.sender), "only seller or active operator");
+        // restrict state machine
+        require(!isDeposited() || Countdown.isOver(), "only after deposit and countdown");
```

Comment:
Should this be `isDeposited` instead of `!isDeposited`? I think we need to re-examine all the states that cancel is valid, since it's a bit of a mess right now.

Reply:
Agreed need to revisit states + split into separate functions

Reply:
fixed in 50df9d5